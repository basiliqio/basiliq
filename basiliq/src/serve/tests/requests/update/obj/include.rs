use super::*;

lazy_static::lazy_static! {
    pub static ref UPDATE_BODY: serde_json::Value =
    json!({
        "data": json!({
            "type": "public__peoples",
            "id": PEOPLES_IDS[0],
            "attributes": json!({
                "age": 100
            })
        })
    });
}

crate::run_test_request!(
    include_multi_rel_m2m,
    Method::PATCH,
    format!(
        "/public__peoples/{}?include=public__people-article",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    include_multi_rel_m2o,
    Method::PATCH,
    format!(
        "/public__peoples/{}?include=public__articles",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    include_single_rel,
    Method::PATCH,
    format!(
        "/public__peoples/{}?include=public__favorite_color",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    include_single_rel_sparsing_all,
    Method::PATCH,
    format!(
        "/public__peoples/{}?include=public__favorite_color&fields[public__favorite_color]=color",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);

crate::run_test_request!(
    include_single_rel_sparsing_none,
    Method::PATCH,
    format!(
        "/public__peoples/{}?include=public__favorite_color&fields[public__favorite_color]=",
        PEOPLES_IDS[0]
    ),
    200,
    UPDATE_BODY.clone()
);
