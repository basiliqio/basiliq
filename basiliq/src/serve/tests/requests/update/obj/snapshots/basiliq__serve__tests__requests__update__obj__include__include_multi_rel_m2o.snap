---
source: basiliq/src/serve/tests/requests/update/obj/include.rs
expression: res

---
{
  "jsonapi": {
    "version": "1.0"
  },
  "data": {
    "type": "public__peoples",
    "id": "[uuid]",
    "attributes": {
      "age": 100,
      "first-name": "Francis",
      "gender": "M",
      "last-name": "Le Roy",
      "twitter": null
    },
    "relationships": {
      "public__articles": {
        "data": [
          {
            "type": "public__articles",
            "id": "[uuid]"
          },
          {
            "type": "public__articles",
            "id": "[uuid]"
          }
        ]
      }
    }
  },
  "included": [
    {
      "type": "public__articles",
      "id": "[uuid]",
      "attributes": {
        "body": "Yeah I know ! Right ?!",
        "title": "Oh my g**"
      }
    },
    {
      "type": "public__articles",
      "id": "[uuid]",
      "attributes": {
        "body": "Yes",
        "title": "How to dead"
      }
    }
  ]
}
